conf/main/grakn.properties

```diff
@@ -16,122 +16,224 @@
 # along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
 #
 
-# Internal Factory Definition
+############################# Graph Configuration #############################
+
+# The factory determines which graph backend Grakn should use. Out of the box,
+# Grakn provides ai.grakn.factory.{TitanInternalFactory, TinkerInternalFactory}.
+#
+# - TitanInternalFactory is a factory providing Titan graphs, that can be configured
+#   to run on different database storage backends (see Titan configuration section).
+#   Grakn recommends using TitanInternalFactory and cassandra for all production
+#   applications.
+# - TinkerInternalFactory is a factory providing TinkerGraphs, in-memory non-persisted
+#   graphs. Does not support transactions and due to lack of indexing, cannot scale.
+#   This factory should be used for testing or for toy examples, never in
+#   production.
 factory.internal=ai.grakn.factory.TitanInternalFactory
+# Factory that provides graphs to the analytics component. Graphs provided by this
+# factory are not used for querying or building graphs. By default, Grakn supports
+# ai.grakn.factory.{TitanHadoopInternalFactory, TinkerInternalFactory}.
+#
+# - TitanHadoopInternalFactory provides a wrapper around a hadoop graph. Wraps around
+#   a gremlin HadoopGraph (IS THIS TRUE)
+# - TinkerInternalFactory is a factory providing TinkerGraphs, in-memory non-persisted
+#   graphs. Does not support transactions and due to lack of indexing, cannot scale.
+#   This factory should be used for testing or for toy examples, never in
+#   production.
 factory.analytics=ai.grakn.factory.TitanHadoopInternalFactory
 
 # Computer
+# This is being removed in another PR
 graph.computer=ai.grakn.graph.internal.computer.GraknSparkComputer
 
-#In-Memory or Distributed task manager
-taskmanager.implementation=ai.grakn.engine.tasks.manager.StandaloneTaskManager
+# The name of the default graph keyspace. This is mainly used
+graph.default-keyspace=grakn
 
-# Logging
-log.dirs=../logs/
-log.level=INFO
+# Number of instances after which Grakn should shard any type node. Sharding is the
+# mechanism by which Grakn mitigates against supernodes.
+# A larger sharding threshold will increase runtime as the graph grows and decrease
+# the likelihood of supernodes. A threshold that is too small will create supernodes
+# more frequently.
+graph.sharding-threshold=10000
 
-#Spark Config
+# How long, in milliseconds, graph-level ontology elements will be
+# cached. Longer cache timeouts means writing will be much faster, particularly for
+# batch loading. Shorter cache timeouts are better for memory usage and in some cases
+# may help avoid GC issues.
+graph.ontology-cache-timeout-ms=600000
+graph.batch.ontology-cache-timeout-ms=600000
+
+############################# Server Configuration #############################
+
+# Port number to use for Grakn engine server to listen on. If provided port = 0
+# then an arbitrary available port will be used
```

Comment:
Its actually because we are calling an underlying spark library with the value of this key and the spark library arbitrarily assigns a port when the value is 0. Do you think we should not allow that and say the port must be a 4 digit number?